include/joystick.h

```diff
@@ -0,0 +1,52 @@
+#ifndef JOYSTICK_H
+#define JOYSTICK_H
+#include "button_state.h"
+
+#include <Arduino.h>
+
+enum class JoystickPosition {
```

Comment:
For the `Button` class, the enum with states is in a separate file. Might be better to have that for the joystick as well